src/server/scripts/Zandalar/KingsRest/kings_rest.cpp

```diff
@@ -139,11 +147,116 @@ class conversation_kings_rest_intro : public ConversationScript
     EventMap _events;
 };
 
+enum AnimatedGuardianEvent
+{
+    EVENT_SUPPRESSION_SLAM = 1
+};
+
+// 133935 - Animated Guardian
+struct npc_kings_rest_animated_guardian : public ScriptedAI
+{
+    npc_kings_rest_animated_guardian(Creature* creature) : ScriptedAI(creature), _suppressionSlamCasts(0) { }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        Talk(SAY_ANIMATED_GUARDIAN_ENGAGE);
+        _events.ScheduleEvent(EVENT_SUPPRESSION_SLAM, 12s);
+    }
+
+    void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
+    {
+        if (me->HealthBelowPctDamaged(50, damage))
```

Comment:
This will trigger any time the creature receives ANY damage below 50%. Is it supposed to be a unique thing?